subsys/bluetooth/services/dis.c

```diff
@@ -45,105 +48,265 @@ static struct dis_pnp dis_pnp_id = {
 };
 #endif
 
+#if defined(CONFIG_BT_DIS_SYSTEM_ID)
+/*
+ * Casting to uint64_t since the value is at most 5 bytes, but will appear as a 32-bit literal if it
+ * is less, giving a warning when right-shifting by 32.
+ */
+static uint8_t dis_system_id[8] = {
+	BT_BYTES_LIST_BE24(CONFIG_BT_DIS_SYSTEM_ID_OUI),
+	BT_BYTES_LIST_BE40((uint64_t)CONFIG_BT_DIS_SYSTEM_ID_IDENTIFIER)};
```

Comment:
Is the `uint64_t` cast necessary?

Reply:
Yes. If you choose a value less than 2^32, then it will give a warning when left-shifting by 32.